Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,248 @@
+                Push(pointLow, ref _pivotPoints);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(pointLow));
+
+                return 2m;
+            }
+
+            if (isHigh)
+            {
+                var point = new PivotPoint(PivotPointType.High, middlePoint.High, middlePoint.EndTime);
+                Push(point, ref _pivotPoints);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(point));
+
+                return 1m;
+            }
+
+            if (isLow)
+            {
+                var point = new PivotPoint(PivotPointType.Low, middlePoint.Low, middlePoint.EndTime);
+                Push(point, ref _pivotPoints);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(point));
+
+                return -1m;
+            }
+
+            return 0m;
+        }
+
+        /// <summary>
+        /// Get current high pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of high pivot points.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetHighPivotPointsArray()
+        {
+            return _pivotPoints?.Where(p => p.PivotPointType == PivotPointType.High).ToArray();
+        }
+
+        /// <summary>
+        /// Get current low pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of low pivot points.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetLowPivotPointsArray()
+        {
+            return _pivotPoints?.Where(p => p.PivotPointType == PivotPointType.Low).ToArray();
+        }
+
+        /// <summary>
+        /// Get all pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of low and high pivot points. Ordered by time in descending order.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetAllPivotPointsArray()
+        {
+            // Get the copy of array
+            return _pivotPoints?.ToArray();
+        }
+
+        /// <summary>
+        /// Invokes NewPivotPointFormed event
+        /// </summary>
+        /// <param name="pivotPointsEventArgs"></param>
+        protected virtual void OnNewPivotPointFormed(PivotPointsEventArgs pivotPointsEventArgs)
+        {
+            NewPivotPointFormed?.Invoke(this, pivotPointsEventArgs);
+        }
+
+        // Will replace an array passed by reference with a new one, containing new pivot point,
+        // where the the order of the elements will be from nearest to farthest in time
```

Comment:
- Does this grow forever in size? If so let's make it capped to save ram.
 - I think you're describing a rolling window here? `RollingWindow`